### DISCUSSION TITLE
MacOS - Disconnect/Reconnect

### DISCUSSION THREAD

**Discussion by linickx** [cgiesche/streamdeck-homeassistant#266] [Help and Support]
I've been using this plug-in for a very long time, but have noticed that recently, I now have to reconnect every morning, this button...

![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/649661/842fb6b1-9507-4bd1-97d5-fbbe523371b6)

Once I click it, everything works fine.

I have the plug-in installed on 2x computers, windows & macOS, this issue only happens on MacOS. I suspect that either a recent update to MacOS or StreamDeck is the cause, but I wonder if anything can be done on the plug-in side; t[he Native HA MacOS app has a similar issue](https://github.com/home-assistant/frontend/issues/17409).

- Is this impacting others? 
- Is it possible to implement some kind of keep-alive, or a trigger to automatically/periodically reconnect?

StreamDeck: 6.5.2 (19936)
HA Plugin: 2.4.0
MacOS: 13.6.6 (Arm/Apple Silicon)
HA Core: 2024.2.3

**Comment by nahog** [comment]
+1

I think there is something that is preventing to reconnect on certain conditions, but that for sure happens when the StreamDeck app starts.

As something similar also happens to me, I though it was because my computer went to sleep, but I tested just by putting my computer to sleep and waking it up and I cannot reproduce it, but for sure it happens every morning.

For me the "reconnect" is not always an option (I had to recheck when it works and when do not) but I need to close the StreamDeck app on Windows and relaunch it for home assistant to connect again.

StreamDeck: 6.6.0
HA Plugin: 2.4.5
Windows 11: 23H2 
HA Core: 2024.4.0
